Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,559 @@
+        m_isUPpressed = true;
+      end
+
+      if (uiKey == Keys.D) then
+        keyPanChanged = true;
+        m_isRIGHTpressed = true;
+      end
+
+      if (uiKey == Keys.S) then
+        keyPanChanged = true;
+        m_isDOWNpressed = true;
+      end
+
+      if (uiKey == Keys.A) then
+        keyPanChanged = true;
+        m_isLEFTpressed = true;
+      end
+    end
+  end
+
+  if (keyPanChanged == true) then
+    -- Base game file uses m_edgePanX and m_edgePanY... but those values do not appear to ever change from 0, so just send 0.
+    ProcessPan(0, 0);
+    return true;
+  end
+
+  -- Allow other handlers of KeyDown events to capture input, if any exist after World Input
+  return false;
+end
+
+-- ===========================================================================
+function DefaultKeyUpHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyUpHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+
+  -- If set to Standard, just let the game do its thing
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_STANDARD then
+    if uiKey == Keys.VK_SHIFT then
+      CQUI_isShiftDown = false;
+    end
+  
+    if uiKey == Keys.VK_ALT then
+      CQUI_isAltDown = false;
+    end
+
+    return BASE_CQUI_DefaultKeyUpHandler(uiKey);
+  end
+
+  local keyPanChanged  :boolean = false;
+  local cquiHandledKey :boolean = false;
+
+  local selectedUnit = UI.GetHeadSelectedUnit();
+  local unitType = nil;
+  local formationClass = nil;
+  if (selectedUnit) then
+    unitType = GameInfo.Units[selectedUnit:GetUnitType()].UnitType;
+    formationClass = GameInfo.Units[selectedUnit:GetUnitType()].FormationClass;
+  end
+
+  --CQUI Keybinds
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+    if (uiKey == Keys.W) then
+      m_isUPpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.D) then
+      m_isRIGHTpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.S) then
+      m_isDOWNpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.A) then
+      m_isLEFTpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.E) then
+      if (CQUI_isAltDown == true) then
+        -- Behave as if just "E" was pressed, send the unit exploring
+        -- Calling this using UnitOperationTypes doesn't work as Automate_Explore hash does not appear among the GameInfo.OperationTypes
+        UnitManager.RequestOperation(UI.GetHeadSelectedUnit(), GameInfo.UnitOperations["UNITOPERATION_AUTOMATE_EXPLORE"].Hash);
+        -- Use false as we need to let the unmodifed script deal with the Alt Key
+        cquiHandledKey = false;
+      elseif (CQUI_cityview) then
+        LuaEvents.CQUI_GoNextCity();
+        cquiHandledKey = true;
+      else
+        UI.SelectNextReadyUnit();
+        cquiHandledKey = true;
+      end
+    end
+
+    if (uiKey == Keys.Q) then
+      if (CQUI_isAltDown == true and unitType == "UNIT_BUILDER" ) then
```

Comment:
I think old version would allow Alt+Q for all hotkey modes except standard. Have to take care it doesn't cause map move + build quarry if you do relocate it.

Reply:
Good catch - I think I intended to move the Alt-Q handler and forgot to put it back.